### PR TITLE
fix: Client crashing on firefox private window

### LINKED ISSUE #32758: RC 6.10.0 WebUI will not open in FireFox private-mode
### Description:

Since upgrading to RC version 6.10.0 today, RC refuses to open in FireFox browser (private mode).
Just the three dots are running forever and no login-screen is displayed.

Via the developer-console the following error can be seen:
**Uncaught TypeError: navigator.serviceWorker is undefined**
Followed by a bunch of  *.ts / *.js resources

After some google work, I found this is a bug that occurs depending on how "navigator.serviceWorker" is defined.
Explictely seems to be an issue with FireFox private mode. In normal mode everythin works as expected.

In the previous releases, private mode of FireFox worked fine.

### Steps to reproduce:

1. Update fo RC version 6.10.0
2. Set FireFox to "always use private mode" 
3. Reload the page and the error occurs

### Expected behavior:

RocketChat WebUI should work in FireFox private mode.

### Actual behavior:

As stated above, since RC 6.10.0 WebUI doesn't seem to work in FireFox private-mode.
Details above.

### Server Setup Information:

- Version of Rocket.Chat Server: 6.10.0
- License Type: Starter
- Number of Users: 10
- Operating System: Debian Bookworm (well patched)
- Deployment Method: docker
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version: (whatever is built in the official docker-container)
- MongoDB Version: 5.0

### Client Setup Information

- FireFox 128.0 (64 bit)
- Operating System: Windows 10, but reproducable on Windows 11 and Linux

### Relevant logs:

Browser-Error-Message:
```
Uncaught TypeError: navigator.serviceWorker is undefined
    ts useDownloadFromServiceWorker.ts:10
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    tsx AttachmentEncryptedDownload.tsx:2
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    tsx AttachmentDownload.tsx:2
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    tsx MessageCollapsible.tsx:1
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    tsx AudioAttachment.tsx:2
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    tsx FileAttachment.tsx:1
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    tsx AttachmentsItem.tsx:2
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    tsx Attachments.tsx:3
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    tsx QuoteAttachment.tsx:2
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    tsx ForwardMessageModal.tsx:2
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    ts messageActionDefault.ts:2
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    ts index.ts:1
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    ts accounts.ts:1
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    o 949676bf261dba48b3da60a41c67af2933e3a64b.js:5
    ts main.ts:1
    g 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    require 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    t 949676bf261dba48b3da60a41c67af2933e3a64b.js:3
    <anonymous> transforms.ts:24
```

### LINKED ISSUE #32826: There is no route for the path: / when trying to deploy rocketchat 6.10.0 on an air-gapped environment
### Description:

When trying to deploy rocket-chat version 6.10.0 on an air-gapped environment the website is not loading.
The loading animation is running infinitely. 
The console shows this error : There is no route for the path: /

This case does not happen on version 6.8.3 or 6.9.3 which works as expected.

### Steps to reproduce:

1. Perform a fresh installation (using the docker installation)
2. Try to navigate into http://<IP>:3000
3. See the error

### Expected behavior:

The first usage wizard should be displayed

### Actual behavior:

rocket-chat

### Server Setup Information:

- Version of Rocket.Chat Server: 6.10.0
- Operating System: Ubuntu Server 22.04
- Deployment Method: docker
- Number of Running Instances: 1
- DB Replicaset Oplog: default
- NodeJS Version: default
- MongoDB Version: default

### Client Setup Information

- Desktop App or Browser Version: Chrome 126.0.6478.126
- Operating System: Windows/Linux

### PR DESCRIPTION
**Pull request by yash-rajpal**
## Proposed changes (including videos or screenshots)

The web client was crashing on firefox's private window, due to unavailability of service workers on private window.

This PR handles this case, the limitation being that users cannot view or download the encrypted attachments on firefox private window. 
## Issue(s)

Closes #32758 
Closes #32826 
## Further comments

https://rocketchat.atlassian.net/browse/CORE-559

### CODE REVIEW

File: .changeset/new-balloons-speak.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@rocket.chat/meteor': patch
+---
+
+Fixed web client crashing on firefox private window. We use serviceWorkers for end to end encrypted file messages, this service worker is not available for firefox private window, so handled that.
```
Comment (KevLehman) [inline .changeset/new-balloons-speak.md:5]:
```suggestion
Fixed web client crashing on Firefox private window. Firefox disables access to service workers inside private windows. Rocket.Chat needs service workers to process E2EE encrypted files on rooms. These types of files won't be available inside private windows, but the rest of E2EE encrypted features should work normally
```

wdyt?

**Review by MartinSchoeler — CHANGES_REQUESTED** [review]
As mentioned via chat, the channel files tab is crashing the browser too, we need to check if there any other places that tries to access the problematic property

**Review by MartinSchoeler — APPROVED** [review]

**Pull request merged**